pkg/destroy/libvirt/libvirt_prefix_deprovision.go

```diff
@@ -239,3 +243,23 @@ func deleteNetwork(conn *libvirt.Connect, filter filterFunc, logger log.FieldLog
 	}
 	return true, nil
 }
+
+// NewLibvirtDestroyer returns libvirt Uninstaller from ClusterMetadata.
+func NewLibvirtDestroyer(level log.Level, metadata *types.ClusterMetadata) (destroy.Destroyer, error) {
+	return &ClusterUninstaller{
+		LibvirtURI: metadata.ClusterPlatformMetadata.Libvirt.URI,
+		Filter:     AlwaysTrueFilter(), //TODO: change to ClusterNamePrefixFilter when all resources are prefixed.
+		Logger: log.NewEntry(&log.Logger{
+			Out: os.Stdout,
+			Formatter: &log.TextFormatter{
+				FullTimestamp: true,
+			},
+			Hooks: make(log.LevelHooks),
+			Level: level,
+		}),
+	}, nil
+}
+
+func init() {
```

Comment:
can we move these `init` to top; better visibility?

Reply:
> can we move these `init` to top; better visibility?

Where do you want them?  It feels odd to put them above the package-level `type` and `var`, although I can do that if you want it.  If I make it the first `func`, it would come between `type ClusterUninstaller struct` and `func (o *ClusterUninstaller) Run()`, which also feel strange.  Then there are a bunch of `Run` helpers.  By the time we get to somewhere that seems like a reasonable spot, we're pretty much at the end of the file.  So I'm fine moving it, but want you to tell me where you'd like it more specifically ;).

Reply:
in that case separate file `register.go` in this package.

Also how do you ensure `Registry` is initialized ? setting directly seems racey?

Reply:
> in that case separate file `register.go` in this package.

Done in e65f8f7 -> 70a37ae.  I also pulled the AWS registration out into `pkg/destroy/aws.go`.

> Also how do you ensure `Registry` is initialized ? setting directly seems racey?

I've adjusted this to use `make()` when I declare the variable.